runtime/common/marshaling/index.ts

```diff
@@ -158,16 +173,30 @@ function isMarshaledValue(val: any): boolean {
   return typeof val === 'object' && MarshalingInjectedKeys.CodaMarshaler in val;
 }
 
-export function marshalValue(val: any): MarshaledValue {
+function marshalValueInternal(val: any, forLogging: boolean): MarshaledValue {
   const postTransforms: PostTransform[] = [];
-  const {val: encodedVal} = fixUncopyableTypes(val, [], postTransforms, 0);
+  const {val: encodedVal} = fixUncopyableTypes(val, [], postTransforms, forLogging, 0);
   return {
     encoded: encodedVal,
     postTransforms,
     [MarshalingInjectedKeys.CodaMarshaler]: CodaMarshalerType.Object,
   };
 }
 
+// The marshaling for console.log is willing to be more lossy/forgiving than the marshaling
+// for other functions since you usually want console.log to print _something_ rather than
+// throw an error.
+//
+// Currently only functions and promises are detected and turned into strings. Other unsupported
+// types could still raise an error.
+export function marshalValueForLogging(val: any): MarshaledValue {
+  return marshalValueInternal(val, true);
```

Comment:
assuming `util.format` is browserified. maybe this can simply be `return format(val)`;